tests/bsim/bluetooth/mesh/src/test_advertiser.c

```diff
@@ -282,26 +237,6 @@ static void xmit_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type
 	}
 }
 
-static void rx_xmit_adv(void)
```

Comment:
Why did you move `rx_xmit_adv` in common place but left `rx_gatt_beacons` and `receive_order`?
It should be unified in general manner or shouldn't be touched at all. Right now, the same behavior is located in two different files: in general file and in test scenario implementation.

It is better to concentrate on your target rather to start refactoring of another tests in the same PR that devoted to suspend resume functionality.

Reply:
The implementation of `receive_order` is tailored to the adv test with number of expected advs, but I agree that `rx_gatt_beacons` could be refactored to something like "`bt_mesh_test_wait_for_adv(&observer_sem, gatt_scan_cb)`". Any thoughts on this?

The thought was to extract the functionality that would be duplicate between the two tests, but it seems that there was more duplicate functionality than expected.

Reply:
I agree, I will address the other comments before looking into adding a separate commit with more refactoring :)

Reply:
I'm now using `bt_mesh_test_wait_for_adv` in all places we have discussed, and generalized it to having variable waiting time.

Reply:
Yep, currently refactoring of the scanner related code looks much better.